gaiku-amethyst/examples/terrain.rs

```diff
@@ -0,0 +1,197 @@
+        )
+        .with_plugin(RenderShaded3D::default())
+        .with_plugin(RenderUi::default())
+        .with_plugin(RenderSkybox::with_colors(
+            Srgb::new(0.82, 0.51, 0.50),
+            Srgb::new(0.18, 0.11, 0.85),
+        ));
+
+    let game_data = GameDataBuilder::default()
+        .with_bundle(render_bund)?
+        // With transform systems for position tracking
+        .with_bundle(TransformBundle::new())?
+        .with_bundle(FlyControlBundle::<StringBindings>::new(
+            Some(String::from("right")),
+            Some(String::from("up")),
+            Some(String::from("forward")),
+        ))?
+        .with_bundle(input_bundle)?
+        .with_bundle(UiBundle::<StringBindings>::new())?;
+
+    let mut game = Application::new(assets_dir, GameLoad::new(), game_data)?;
+
+    game.run();
+    Ok(())
+}
+
+pub struct GameLoad;
+
+impl SimpleState for GameLoad {
+    fn on_start(&mut self, data: StateData<'_, GameData<'_, '_>>) {
+        let world = data.world;
+
+        self.initialise_camera(world);
+        self.add_light(world);
+        self.build_terrain(world);
+    }
+}
+
+impl Default for GameLoad {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl GameLoad {
+    pub fn new() -> Self {
+        Self {}
+    }
+
+    fn initialise_camera(&self, world: &mut World) {
+        let mut transform = Transform::default();
+        transform.set_translation_xyz(0., 2., -10.0);
+        transform.face_towards(Vector3::new(0., 0., 0.), Vector3::new(0., 1., 0.));
+
+        let cam_ent = world
+            .create_entity()
+            .with(Camera::standard_3d(600., 400.))
+            .with(transform)
+            .with(FlyControlTag)
+            .build();
+        let act_cam: &mut ActiveCamera = world.get_mut().expect("There shoud be an active camera");
+        act_cam.entity = Some(cam_ent);
+    }
+
+    fn add_light(&self, world: &mut World) {
+        world
+            .create_entity()
+            .with(Light::from(DirectionalLight {
+                color: Rgb::new(1.0, 1.0, 1.0),
+                direction: [-1.0, -1.0, -1.0].into(),
+                intensity: 1.0,
+            }))
+            .build();
+    }
+
+    fn build_terrain(&self, world: &mut World) {
+        let file = format!(
+            "{}/examples/assets/{}.gox",
+            env!("CARGO_MANIFEST_DIR"),
+            "terrain"
+        );
+        let chunks = GoxReader::read(&file);
+        let mut meshes = vec![];
+
+        for chunk in chunks.iter() {
+            let mesh = VoxelBaker::bake(chunk);
+            if let Some(mesh) = mesh {
+                meshes.push((mesh, chunk.position()));
+            }
+        }
+
+        let scale = Vector3::new(0.1, 0.1, 0.1);
+        let swap_axes = true;
+        let transform = if swap_axes {
+            Matrix4::from_euler_angles(-90., 0., 0.)
+        } else {
+            Matrix4::identity()
+        };
```

Comment:
Rotation is done here. To swap the y and z axes.

Reply:
Yeah, checked now and seems that it works as expected in master.